packages/aws-cdk-lib/aws-sns/lib/subscription.ts

```diff
@@ -136,6 +144,43 @@ export class Subscription extends Resource {
       throw new Error('Subscription role arn is required field for subscriptions with a firehose protocol.');
     }
 
+    if (props.deliveryPolicy) {
+      if (props.protocol !== SubscriptionProtocol.HTTP && props.protocol !== SubscriptionProtocol.HTTPS) {
+        throw new Error('Delivery policy is only supported for HTTP and HTTPS subscriptions');
+      }
+      const { healthyRetryPolicy, throttlePolicy } = props.deliveryPolicy;
+      if (healthyRetryPolicy) {
+        const delayTargetLimit = 3600;
+        if (healthyRetryPolicy.minDelayTarget < 1 || healthyRetryPolicy.minDelayTarget > delayTargetLimit) {
+          throw new Error(`minDelayTarget must be between 1 and ${delayTargetLimit} inclusive`);
+        }
+        if (healthyRetryPolicy.maxDelayTarget < 1 || healthyRetryPolicy.maxDelayTarget > delayTargetLimit) {
+          throw new Error(`maxDelayTarget must be between 1 and ${delayTargetLimit} inclusive`);
+        }
+        if (healthyRetryPolicy.minDelayTarget > healthyRetryPolicy.maxDelayTarget) {
+          throw new Error('minDelayTarget must not exceed maxDelayTarget');
+        }
+        const numRetriesLimit = 100;
+        if (healthyRetryPolicy.numRetries < 0 || healthyRetryPolicy.numRetries > numRetriesLimit) {
```

Comment:
Is it necessary to validate that numRetries is an Integer?

Reply:
Yeah good catch

Reply:
Has the integer check been added?

Reply:
Sorry, forgot to push a commit :sweat_smile: